### ISSUE TITLE
Add shared Entity:AddToMotionController( physobj ) , Entity:RemoveFromMotionController( physobj )

### ISSUE THREAD

**Issue by Jvs34** [Facepunch/garrysmod-requests#424]
This would allow the control of physobjects properly like valve does with the gravity gun and the physics gun instead of having to use hacky stuff like a Think hook ( which results in jaggy movement ).

Yes I know that this is pretty much how it's implemented already, but it's a really selfish implementation, since the entity only cares about its own physobj which makes the current physicssimulate hook unreliable to control other physobjects with.

Also I guess you would also need to remove the check that doesn't create the motion controller if the scripted entity doesn't have its own physobj so we can have a fully invisible controller with no physics.

Here's some code of the implementation I'm asking you for, I guess this is similar to how it's done already, the Lua side is pretty easy I guess.

```
void MyScriptedEntity:CreateMotionController()
{
	if ( m_mymotioncontroller )
	{
		physenv->DestroyMotionController( m_mymotioncontroller );
		m_mymotioncontroller = NULL;
	}
	
	m_mymotioncontroller = physenv->CreateMotionController( whoever_yourlistener_is );
	
	IPhysicsObject * myphysobj = VPhysicsGetObject();
	
	if ( myphysobj )
		AddToMotionController( myphysobj );
}

void MyScriptedEntity:AddToMotionController( IPhysicsObject * physobj )
{
	if ( m_mymotioncontroller && physobj )
	{
		m_mymotioncontroller->AttachObject( physobj, true );
	}
}

void MyScriptedEntity:RemoveFromMotionController( IPhysicsObject * physobj )
{
	if ( m_mymotioncontroller && physobj )
	{
		m_mymotioncontroller->DetachObject( physobj );
	}
}
```

This implementation would be left backwards-compatible with the old one and shouldn't break anything.

**TL;DR: this allows other physobjs to get called in the PhysicsSimulate hook without the need of having a valid physobj on the main Lua entity, so we can do some the same shit as the gravity gun does properly.**

**Comment by robotboy655** [comment]
DONE

**Comment by meepdarknessmeep** [comment]
:clap:

**Comment by xaviergmail** [comment]
:+1:

**Comment by Noiwex** [comment]
Nigger
